catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/services/pipeline/PipelineServiceResource.java

```diff
@@ -0,0 +1,202 @@
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import org.openmetadata.catalog.api.services.CreatePipelineService;
+import org.openmetadata.catalog.api.services.UpdatePipelineService;
+import org.openmetadata.catalog.entity.data.Dashboard;
+import org.openmetadata.catalog.entity.services.MessagingService;
+import org.openmetadata.catalog.entity.services.PipelineService;
+import org.openmetadata.catalog.jdbi3.PipelineServiceRepository;
+import org.openmetadata.catalog.resources.Collection;
+import org.openmetadata.catalog.security.CatalogAuthorizer;
+import org.openmetadata.catalog.security.SecurityUtil;
+import org.openmetadata.catalog.type.EntityReference;
+import org.openmetadata.catalog.util.RestUtil;
+import org.openmetadata.catalog.util.ResultList;
+
+import javax.validation.Valid;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriInfo;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+@Path("/v1/services/pipelineServices")
+@Api(value = "Pipeline service collection", tags = "Services -> Pipeline service collection")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@Collection(name = "pipelineServices", repositoryClass = "org.openmetadata.catalog.jdbi3.PipelineServiceRepository")
+public class PipelineServiceResource {
+  private final PipelineServiceRepository dao;
+  private final CatalogAuthorizer authorizer;
+
+  public static EntityReference addHref(UriInfo uriInfo, EntityReference service) {
+    return service.withHref(RestUtil.getHref(uriInfo, "v1/services/pipelineServices/", service.getId()));
+  }
+
+  private static List<PipelineService> addHref(UriInfo uriInfo, List<PipelineService> instances) {
+    instances.forEach(i -> addHref(uriInfo, i));
+    return instances;
+  }
+
+  private static PipelineService addHref(UriInfo uriInfo, PipelineService pipelineService) {
+    pipelineService.setHref(RestUtil.getHref(uriInfo, "v1/services/pipelineServices/",
+            pipelineService.getId()));
+    return pipelineService;
+  }
+
+  @Inject
+  public PipelineServiceResource(PipelineServiceRepository dao, CatalogAuthorizer authorizer) {
+    Objects.requireNonNull(dao, "PipelineServiceRepository must not be null");
+    this.dao = dao;
+    this.authorizer = authorizer;
+  }
+
+  static class PipelineServiceList extends ResultList<PipelineService> {
+    PipelineServiceList(List<PipelineService> data) {
+      super(data);
+    }
+  }
+
+  @GET
+  @Operation(summary = "List Pipeline services", tags = "services",
```

Comment:
nit: List pipeline services (no need to capitalize p)

Reply:
Fixed